constraints/match.js

```diff
@@ -111,5 +137,93 @@ function hasMatch(sNode, pTree)
 		}
 		return false;
 	}
-	
+
+}
+
+// Match Max constraints:
+
+/* Same as hasMatch function above, except this only returns true if the
+ * matching prosodic node is maximal:
+ */
+function hasMaxMatch(sNode, pTree){
+	 var sLeaves = getLeaves(sNode);
+	 markMinMax(pTree); //mark min and max on prosodic tree
+	 if(catsMatch(sNode.cat, pTree.cat) && sameIds(getLeaves(pTree), sLeaves) && pTree.isMax)
+	 // the current prosodic node is the match, both for category and for terminals, and is maximal
+		{
+ 			return true;
+ 		}
+
+ 		// If the current prosodic node is NOT the match:
+
+ 		else if(!pTree.children || pTree.children.length===0){
+ 		// current node is terminal
+ 			return false;
+		}
+
+ 		else
+ 		//the current prosodic node is non-terminal (has children)
+ 		{
+ 			for(var i = 0; i < pTree.children.length; i++)
+ 			//check each child to see if the match exists in the subtree rooted in that child
+ 			{
+ 				var child = pTree.children[i];
+ 				if(hasMaxMatch(sNode, child)){
+ 					return true;
+				}
+ 			}
+ 			return false;
+ 		}
+
+ }
+
+ /* Assign a violation for every node of syntactic category s that is not
```

Comment:
Just for clarification, please add the following to the comment here:
Match-SP(scat-max, pcat-max)
ex. Match a maximal xp with a maximal phi

Reply:
added in latest commit